### PR TITLE
installation: upgrade recipes fix

### PR DESCRIPTION
**Pull request by egabancho**

### CODE REVIEW

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py [hunk 1]
```diff
@@ -31,46 +33,57 @@ def info():
...
+            db.Column('is_internal', db.Boolean(), nullable=True),
+            db.Column('_redirect_uris', db.Text(), nullable=True),
+            db.Column('_default_scopes', db.Text(), nullable=True),
+            db.ForeignKeyConstraint(['user_id'], ['user.id'], ),
+            db.PrimaryKeyConstraint('client_id'),
+            mysql_charset='utf8',
+            mysql_engine='MyISAM'
+        )
+    else:
+        warnings.warn("*** Creatrion of table 'oauth2CLIENT' skipped!")
```
Comment (jmartinm) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py]: typo

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py [hunk 2]
```diff
@@ -31,46 +33,57 @@ def info():
...
+            db.ForeignKeyConstraint(['client_id'], ['oauth2CLIENT.client_id'], ),
+            db.ForeignKeyConstraint(['user_id'], ['user.id'], ),
+            db.PrimaryKeyConstraint('id'),
+            db.UniqueConstraint('access_token'),
+            db.UniqueConstraint('refresh_token'),
+            mysql_charset='utf8',
+            mysql_engine='MyISAM'
+        )
+    else:
+        warnings.warn("*** Creatrion of table 'oauth2TOKEN' skipped!")
```
Comment (jmartinm) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py]: typo ;)

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py (hunk 1 above)
Comment (egabancho) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py]: Ups! ;)

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py (hunk 2 above)
Comment (egabancho) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_02_17_initial.py]: copy paste ejem :P

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py [hunk 3]
```diff
@@ -44,6 +46,9 @@ def do_upgrade():
...
 def estimate():
     """Estimate running time of upgrade in seconds (optional)."""
-    return run_sql(
-        "SELECT COUNT(id) AS ids FROM oauth2TOKEN"
-    )[0][0]
+    try:
+        return run_sql(
+            "SELECT COUNT(id) AS ids FROM oauth2TOKEN"
+        )[0][0]
+    except:
```
Comment (jirikuncar) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py]: Why do you need to catch the exception here?

--
PS: `except Exception:`

**Comment by jirikuncar** [comment]
- [ ] `4: L102 copyright year is outdated, expected 2015 but got 2014`

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py (hunk 3 above)
Comment (egabancho) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py]: If the table is not there it fails to do the estimate

Comment (jirikuncar) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py]: Can you use `has_table` instead?

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py [hunk 4]
```diff
@@ -44,6 +46,9 @@ def do_upgrade():
 
 def estimate():
     """Estimate running time of upgrade in seconds (optional)."""
-    return run_sql(
-        "SELECT COUNT(id) AS ids FROM oauth2TOKEN"
-    )[0][0]
+    try:
```
Comment (jirikuncar) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py]: see https://github.com/inveniosoftware/invenio/pull/2747#discussion_r24232940

File: invenio/modules/oauthclient/upgrades/oauthclient_2014_10_21_encrypted_token_column.py [hunk 5]
```diff
@@ -43,6 +44,9 @@ def do_upgrade():
 
 def estimate():
     """Estimate running time of upgrade in seconds (optional)."""
-    return run_sql(
-        "SELECT COUNT(*) AS ids FROM remoteTOKEN"
-    )[0][0]
+    try:
```
Comment (jirikuncar) [inline invenio/modules/oauthclient/upgrades/oauthclient_2014_10_21_encrypted_token_column.py]: see https://github.com/inveniosoftware/invenio/pull/2747#discussion_r24232940

File: invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py (hunk 3 above)
Comment (egabancho) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py]: Yep!

Comment (egabancho) [inline invenio/modules/oauth2server/upgrades/oauth2server_2014_10_21_encrypted_token_columns.py]: and done!

**Pull request merged**